test/NIFSemanticValidatorSpecs.js

```diff
@@ -0,0 +1,56 @@
+describe('Testing NIF semantic validator service', function() {
+	var nifSemanticValidatorService;
+
+	beforeEach(function () {
+		// Use the function directly but we are not simulating the use of the service by angular
+		//nifSemanticValidatorService = new nifSemanticValidatorService();
+
+		// Module containing the services module
+		module('app.services');
+
+		inject(function(nifSemanticValidationService) {
+			nifSemanticValidatorService = nifSemanticValidationService;
+		});
+	});
+
+	it('Inform a valid DNI', function() {
```

Comment:
It should be something like

describe("A DNI is a sequence of X digits and a single letter")
it("can match with a given algorithm, case A (letter A)")
(... include some more tests)

it("can fail to validate the algorithm")

*****

describe("A NIE is...")
it...